tests/workflows/stream_try_node_annotation/tests/test_workflow.py

```diff
@@ -38,81 +41,140 @@ def test_workflow_stream__happy_path():
     # node initiated events
     node_initiated_events = [e for e in events if e.name == "node.execution.initiated"]
     assert node_initiated_events[0].node_definition == InnerNode
-    assert node_initiated_events[0].model_dump(mode="json")["body"]["node_definition"] == {
+    assert node_initiated_events[0].model_dump(mode="json")["body"][
+        "node_definition"
+    ] == {
         "name": "TryNode",
-        "module": ["tests", "workflows", "stream_try_node_annotation", "workflow", "InnerNode", ADORNMENT_MODULE_NAME],
+        "module": [
+            "tests",
+            "workflows",
+            "stream_try_node_annotation",
+            "workflow",
+            "InnerNode",
+            ADORNMENT_MODULE_NAME,
+        ],
     }
     assert node_initiated_events[0].parent is not None
     assert node_initiated_events[0].parent.type == "WORKFLOW"
-    assert node_initiated_events[0].parent.workflow_definition == StreamingTryExample
+    assert node_initiated_events[
+        0
+    ].parent.workflow_definition == CodeResourceDefinition(
+        **serialize_type_encoder(StreamingTryExample)
+    )
```

Comment:
[nit] I could also see us supporting a static method on `CodeResourceDefinition` to handle this more natively, eg:

```python
CodeResourceDefinition.from(StreamingTryExample)
```

Reply:
oh that's a great idea, and it'd probably be able to be used pre-validation